### PR TITLE
Replace jmustache with anki.template.template.py

### PR DESCRIPTION
**Pull request by hssm**
Implemented [Issue 1837](https://code.google.com/p/ankidroid/issues/detail?id=1837). This commit ports Anki's template parsing module and removes our dependence on jmustache.

Along the way, also fixes:
- [Issue 2544](https://code.google.com/p/ankidroid/issues/detail?id=2544)
- [Issue 1936](https://code.google.com/p/ankidroid/issues/detail?id=1936)
- Fixed the "hint" feature. The hook was throwing an exception.

My tests have been quite comprehensive and covers every template feature described in the manual. If there is anything that breaks, it will be some combination of features nested together or something like that.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 1]
```diff
@@ -951,19 +946,20 @@ public void updateFieldCache(long[] nids) {
 
 
     public HashMap<String, String> _renderQA(Object[] data, String qfmt, String afmt) {
+        long startTime = System.nanoTime();
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: Is this debug code which you forgot to remove?

Comment (hssm) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: It was. I've removed it.

**Comment by timrae** [comment]
Great, thanks a lot for this important patch

**Comment by timrae** [comment]
I think it would be really useful to add some tests if you (or another volunteer) has time

**Comment by hssm** [comment]
I have a test deck that utilizes all template features. I can add it to some directory in the project (`/tools/decks/` maybe?).

The desktop code doesn't have any tests for template rendering so there wasn't any of that to port over.

**Comment by timrae** [comment]
OK sounds good... maybe "/tools/test-decks/" would be slightly more
descriptive though

**Pull request merged**